### PR TITLE
Add custom repository table [SCI-1274]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-1274
https://biosistemika.atlassian.net/browse/SCI-1290

### CODE REVIEW

File: app/models/repository_rows_my_module.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class RepositoryRowsMyModule < ActiveRecord::Base
```
Comment (Ducz0r) [inline app/models/repository_rows_my_module.rb]: I think this class needs to be in singularity, and maybe use some additional checks (see how I've done it in my PR) - https://github.com/Ducz0r/scinote-web/blob/lm-sci-1279/app/models/repository_row_my_module.rb

Comment (okriuchykhin) [inline app/models/repository_rows_my_module.rb]: I think Rails has special rules for naming join models, which represents many to many association and I think both sides of association should be plural

Comment (Ducz0r) [inline app/models/repository_rows_my_module.rb]: I don't know. Maybe we messed up with the rest of the join tables, which are all named e.g. `user_my_modules`, but that translates into ActiveRecord being named `UserMyModule`, which I believe is a good way to name a join model. I mean, I think it's more consistent to keep this table named `repository_row_my_modules`, which will then also mean that the ActiveRecord is named `RepositoryRowMyModule` (with no plural there).

Comment (okriuchykhin) [inline app/models/repository_rows_my_module.rb]: https://stackoverflow.com/questions/11590469/rails-naming-convention-for-join-table

File: app/datatables/repository_datatable.rb [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+                        # repository_cells: :value
+                      ).references(
+                        :repository_columns,
+                        :created_by
+                      )
+                      .where(repository: @repository)
+    if @my_module
+      @assigned_rows = @my_module.repository_rows
+
+      # repository_rows.joins(
```
Comment (Ducz0r) [inline app/datatables/repository_datatable.rb]: Should this be thrown out?

File: app/datatables/repository_datatable.rb [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    end
+  end
+
+  def new_sort_column(item)
+    coli = item[:column].to_i - 1
+    model, column = sortable_columns[sortable_displayed_columns[coli].to_i]
+                    .split('.')
+
+    return model if model == ASSIGNED_SORT_COL
+    col = [model.constantize.table_name, column].join('.')
```
Comment (Ducz0r) [inline app/datatables/repository_datatable.rb]: Indeed, is this meant as `coli`? Or should both names be `col`. Is this function used at all?

File: app/models/my_modules_repository_row.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class MyModulesRepositoryRow < ActiveRecord::Base
```
Comment (Ducz0r) [inline app/models/my_modules_repository_row.rb]: I still think you should change this to `MyModuleRepositoryRow`, just to be consistent with the rest of the codebase, no other reason.

File: app/models/repository_table.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+class RepositoryTable < ActiveRecord::Base
```
Comment (Ducz0r) [inline app/models/repository_table.rb]: Should we rename this to `RepositoryTableState`? I mean, just to point out specifically that this is a table just to store state data.

Comment (okriuchykhin) [inline app/models/repository_table.rb]: Renamed. I've just made the same name as we use for samples

File: app/models/my_modules_repository_row.rb (hunk 4 above)
Comment (okriuchykhin) [inline app/models/my_modules_repository_row.rb]: Renamed

File: app/datatables/repository_datatable.rb (hunk 2 above)
Comment (okriuchykhin) [inline app/datatables/repository_datatable.rb]: Removed

**Comment by Ducz0r** [comment]
Aaalright, start up the engines and merge away! May God have mercy on our souls :pray: :rainbow:.

**Pull request merged**